src/.clang-format

```diff
@@ -1,50 +1,96 @@
+# -*- mode: yaml; -*-
+
+# Goals
+# - have automated formatting to avoid wasted time
+# - minimize diff-sensitivity to things like changing function names (e.g., don’t want whitespace
+#   diffs on all the parameter lines)
+
+# NB: If clang-format can’t parse this file (or doesn’t understand any of the values, it will format
+#     with the default settings. If this happens it’s important to _revert_ the formatting, not just
+#     reformat after fixing it. This is because some formatting changes aren’t reversible, so you
+#     may introduce formatting regressions if you simply reformat.
+
+# TODO: clang-format 17 adds the ability to include a trailing comment on formatting delimiters, so
+#       once we’re using that version
+#
+#     // NB: why we’re disabling
+#     // clang-format off
+#
+#       can become
+#
+#     // clang-format off: why we’re disabling
+
 AccessModifierOffset: -4
-AlignEscapedNewlinesLeft: true
+AlignAfterOpenBracket: AlwaysBreak
+AlignEscapedNewlines: Left
 AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: false
+AllowAllConstructorInitializersOnNextLine: true
 AllowAllParametersOfDeclarationOnNextLine: false
-AllowShortBlocksOnASingleLine: false
-AllowShortFunctionsOnASingleLine: All
-AllowShortIfStatementsOnASingleLine: false
+AllowShortBlocksOnASingleLine: Empty
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortFunctionsOnASingleLine: Inline
+AllowShortIfStatementsOnASingleLine: true # I’d prefer false
+AllowShortLambdasOnASingleLine: All
 AllowShortLoopsOnASingleLine: false
 AlwaysBreakBeforeMultilineStrings: false
 AlwaysBreakTemplateDeclarations: true
+BinPackArguments: false
 BinPackParameters: false
-BreakBeforeBinaryOperators: false
+BreakBeforeBinaryOperators: None # I’d prefer NonAssignment
 BreakBeforeBraces: Linux
```

Comment:
Consider
```
BreakBeforeBraces: Custom
BraceWrapping:
    AfterCaseLabel: true
    AfterClass: true
    AfterControlStatement: MultiLine
    AfterEnum: true
    AfterFunction: true
    AfterNamespace: true
    AfterStruct: true
    AfterUnion: true
    AfterExternBlock: false
    BeforeCatch: false
    BeforeElse: false
    BeforeLambdaBody: false
    BeforeWhile: false
    SplitEmptyFunction: true
    SplitEmptyRecord: true
    SplitEmptyNamespace: true
```

`AfterCaseLabel: true` reflects the fact that braced blocks aren't special in case branches, they just create a scope as they would anywhere else, and should be formatted as such. (This is a rare case but I am opinionated about it.)
`AfterControlStatement: MultiLine` makes it clearer where the boundary between a multiline condition and the body is, while being concise in other cases.
If `AfterFunction` etc. are set to true, then setting `SplitEmptyFunction` etc. to true makes sense for consistency.

Reply:
On second thoughts, the effect of `SplitEmptyFunction: true` on the verbosity of empty constructors might justify setting `SplitEmptyFunction: false`. (Empty functions are rare at top level but not in constructors.)